### PR TITLE
docs: update local run discussion with inner dev loop

### PR DESCRIPTION
**Pull request by sentinel1909**
Good evening,

I updated the "Local Run" section within "Introduction" in the docs to describe that you can use the inner development loop (with cargo watch and flags) when developing locally with Shuttle.

This may not be a good fit, but I thought I'd suggest it and do a PR for consideration :)

(I was quite excited when I realized this could be done...)

~Jeff Mitchell

### CODE REVIEW

**Comment by joshua-mo-143** [comment]
This looks pretty good!

I was just wondering what your opinion is on this but should we have the title as Incremental Compilation? "Inner dev loop" doesn't seem like a term that's used commonly... at least insofar as I know. Pinging @oddgrd for a second opinion

File: introduction/local-run.mdx [hunk 1]
```diff
@@ -14,6 +14,21 @@ project directory, run:
 cargo shuttle run
 ```
 
+### Inner development loop
+The power of the inner development loop can be harnassed when developing locally with Shuttle! Leverage `cargo watch` like this:
```
Comment (oddgrd) [inline introduction/local-run.mdx:18]:
```suggestion
The power of the inner development loop can be harnessed when developing locally with Shuttle! Leverage `cargo watch` like this:
```

**Review by oddgrd — APPROVED** [review]
Thanks Jeff, apologies for the delayed review. I think it's great that we document using  `cargo watch` with shuttle. I'm also not completely convinced about the naming, but I think we can merge this as is and change it in the future if needed.

File: introduction/local-run.mdx (hunk 1 above)
Comment (sentinel1909) [reply introduction/local-run.mdx:18]: Looks great!

**Comment by jonaro00** [comment]
Isn't cargo clippy just a superset of cargo check? (check could be excluded)

**Comment by jonaro00** [comment]
@sentinel1909 I feel like this example + these https://discordapp.com/channels/803236282088161321/1107140940759568384/1118618874020114482 (read context)
make a good starting point for a "Tips and tricks" type of page.

**Comment by sentinel1909** [comment]
Tried to respond to your last comment @jonaro00, but I'm not managing to quote the original message.  I'm not able to follow that Discord link.

**Comment by sentinel1909** [comment]
> Isn't cargo clippy just a superset of cargo check? (check could be excluded)

I looked back at the material in Luca's  book.  You're right, clippy is left to the CI pipeline as the linting step.  The true "inner dev loop" Luca writes about is: cargo watch -x check -x test -x run  In the context of Shuttle it would be cargo watch -x check -x test -x 'shuttle run'

**Comment by ivancernja** [comment]
Hey @sentinel1909, sorry for being a bit late on this one. I've merged @jonaro00's PR that included an additional tip. Do you reckon we can close this one?

context: https://github.com/shuttle-hq/shuttle-docs/pull/149/files

**Comment by sentinel1909** [comment]
Ha! No worries :) Yes, I think it can be closed.

**Pull request closed without merging**